src/SUFST/src/can_publisher.c

```diff
@@ -56,18 +59,19 @@ void queue_send_thread_entry(ULONG input)
 
     // Simulated CAN message
     rtcan_msg_t queue_data;
+    uint16_t debug_index = 0;
 
     while(1){
     
-    // Simulate CAN Message here. TODO: Get input from file.
-
     queue_data.identifier = CAN_DATABASE_PM100_VOLTAGE_INFO_FRAME_ID;
     queue_data.length = CAN_DATABASE_PM100_VOLTAGE_INFO_LENGTH;
 
-    for(int i = 0; i<8; i++)
+    // Parse lookup table of dummy data here and pack it in queue_data
+    for(int i = 0; i < 8; i++)
     {
-        queue_data.data[i] = i;
+        queue_data.data[i] = debug_lookup[debug_index%16 + i];
     }
+    debug_index++;
 
     // Send the data to the queue.
```

Comment:
Queue pointer should points to same queue that is initialised in unpack thread. You may put `unpack_context` structure as argument to `can_publisher_init()`. In old version we put `publisher_context` structure in `unpack_init()`, but now it will be more convenient to to it other way around.
I don't know if that make sense for you. If not, I can commit my changes to this branch to finalise and integrate code with the `can_unpack`.

Reply:
I did this in the last commit but as you can see on the commit message I'm not sure if I'm doing it correctly (it compiles well, but I feel like it won't work as expected) 
Since I don't have STM32 board right now so I can't debug on the actual board, can you check this and fix issues if there're any please?
@rureverek

Reply:
Last commit is not really right, but it's not a problem, I'll fix it.

Reply:
Thanks, I’ll learn something from your code ;)